src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -253,6 +299,38 @@ export class WorkspaceSavedObjectsClientWrapper {
       objects: SavedObjectsBulkGetObject[] = [],
       options: SavedObjectsBaseOptions = {}
     ): Promise<SavedObjectsBulkResponse<T>> => {
+      const workspaceIds: string[] = [];
+      const nonWorkspaceObjects: SavedObjectsBulkGetObject[] = [];
+      objects.forEach((obj) => {
+        if (this.isRelatedToWorkspace(obj.type)) {
+          workspaceIds.push(obj.id);
+        } else {
+          nonWorkspaceObjects.push(obj);
+        }
+      });
+
+      const permittedWorkspaceIds =
+        (await this.permissionControl.getPermittedWorkspaceIds(wrapperOptions.request, [
+          WorkspacePermissionMode.LibraryRead,
+          WorkspacePermissionMode.Management,
+        ])) ?? [];
+      const workspacePermitted = workspaceIds.every((id) => permittedWorkspaceIds.includes(id));
+      if (!workspacePermitted) {
+        throw generateWorkspacePermissionError();
+      }
+
+      // saved_objects
+      const objectsPermitted = await this.permissionControl.batchValidate(
+        wrapperOptions.request,
+        nonWorkspaceObjects.map((savedObj) => ({
+          ...savedObj,
+        })),
+        [WorkspacePermissionMode.Read]
+      );
+      if (!objectsPermitted) {
+        throw generateSavedObjectsPermissionError();
```

Comment:
We don't need to throw error here if not permitted. Because we need to verify the permission of workspaces `attributes` later.